src/python/ksc/cgen.py

```diff
@@ -160,12 +165,23 @@ def arg_types_of_decl(decl):
 
 
 def generate_cpp_entry_point(
-    cpp_function_name: str, decl: Def, vectorization: VecSpec, use_torch: bool
+    cpp_function_name: str,
+    decl: Def,
+    vectorization: VecSpec,
+    use_torch: bool,
+    gpu: bool,
 ):
     if isinstance(vectorization, VecSpec_Elementwise):
         if not use_torch:
             raise ValueError("Elementwise operations only available when using torch")
-        return generate_cpp_elementwise_entry_point(cpp_function_name, decl)
+        if gpu:
+            return generate_cpp_cuda_entry_point(cpp_function_name, decl)
+        else:
+            return generate_cpp_elementwise_entry_point(cpp_function_name, decl)
+    if gpu:
+        raise ValueError(
+            "Only elementwise operations can be compiled for GPU"
```

Comment:
Do you mean, fall back to CPU code in that case?

Reply:
You wouldn't want to do it for performance reasons but could the function still run on GPU?

I'm thinking more of the case of elementwise operations followed by a scalar one. It's not important, it was more for people just trying things out.

Reply:
Ah I see - yes we could run it on a single GPU thread. I'll have a look to see how much extra code we'd need to write to make this work.

Reply:
So I think this is possible but I'm going to suggest not doing it in this PR. It would need another version of the code-generator in `cgen.py`, and we'd also need to add a new example to test it on, so I think it's a large enough addition that it should be a separate change. (Also there still have to be some restrictions on the types used in the function.)